opte-api/src/ip.rs

```diff
@@ -538,6 +546,11 @@ impl Ipv6Addr {
         Self::from_const([0xff02, 0, 0, 0, 0, 1, w0, w1])
     }
 
+    /// Return `true` if this is a solicited node multicast address.
+    pub fn is_solicited_node_multicast(&self) -> bool {
+        &self.inner[..12] == &[0xff, 0x02, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0x01]
```

Comment:
Does this also have to verify the `0xFF` in octet 13?

```
   The above multicast address is computed as a function of a node's
   unicast and anycast addresses.  The solicited-node multicast address
   is formed by taking the low-order 24 bits of the address (unicast or
   anycast) and appending those bits to the prefix
   FF02:0:0:0:0:1:FF00::/104 resulting in a multicast address in the
   range

      FF02:0:0:0:0:1:FF00:0000

   to

      FF02:0:0:0:0:1:FFFF:FFFF
```

Reply:
It does yeah, thanks.